### PR TITLE
Better handling of errors on grunt build checks

### PR DESCRIPTION
**Pull request by noliveleger**
As suggested by @jnm, quick change to handle `sha1sum` errors.

### CODE REVIEW

**Comment by MartijnR** [comment]
best for @jnm to review (also the previous PR that I approved but would have been better checked by @jnm).

File: setup/docker/01_setup_enketo.bash [hunk 1]
```diff
@@ -17,23 +17,33 @@ SHA1SUM_CONFIG_FILE_PATH="${CHECKSUM_DIR_PATH}/${CONFIG_FILE}.sha1"
...
 # Compare config version
-set +e  # Do not exit on `sha1sum` or `cat` error
-sha1sum -c "${SHA1SUM_CONFIG_FILE_PATH}"
-RUN_GRUNT=$([[ $? -eq 0 ]] && echo 0 || echo 1)
+sha1sum -c "${SHA1SUM_CONFIG_FILE_PATH}" || RUN_GRUNT=1
 
 # Compare commit version
 CURRENT_COMMIT=$(git rev-parse HEAD) # Get current commit
-LAST_BUILD_COMMIT=$(cat ${LAST_BUILD_COMMIT_FILE_PATH})
+LAST_BUILD_COMMIT=$(cat ${LAST_BUILD_COMMIT_FILE_PATH}) || echo "-"
```
Comment (jnm) [inline setup/docker/01_setup_enketo.bash]: If you want `-` to end up in echo `LAST_BUILD_COMMIT` if there's a problem reading from `LAST_BUILD_COMMIT_FILE_PATH`, then I think the `|| echo "-"` should be inside the `$()`:
```
john@world$ LAST_BUILD_COMMIT_FILE_PATH=fake
john@world$ LAST_BUILD_COMMIT=$(cat ${LAST_BUILD_COMMIT_FILE_PATH}) || echo "-"
cat: fake: No such file or directory
-
john@world$ echo $LAST_BUILD_COMMIT

john@world$ LAST_BUILD_COMMIT=$(cat ${LAST_BUILD_COMMIT_FILE_PATH} || echo "-")
cat: fake: No such file or directory
john@world$ echo $LAST_BUILD_COMMIT
-
```

File: setup/docker/01_setup_enketo.bash [hunk 2]
```diff
@@ -17,23 +17,33 @@ SHA1SUM_CONFIG_FILE_PATH="${CHECKSUM_DIR_PATH}/${CONFIG_FILE}.sha1"
...
 RUN_GRUNT=$([[ "$RUN_GRUNT" == 0 && "$CURRENT_COMMIT" == "$LAST_BUILD_COMMIT" ]] && echo 0 || echo 1)
 
-set -e  # Restore exit mode
+# Ensure build files exist.
+# Can't test only on the parent folder because it's created by docker
+NOT_EMPTY_JS_PATH="public/js/build/.not-empty"
+NOT_EMPTY_CSS_PATH="public/css/.not-empty"
+NOT_EMPTY_LOCALES_PATH="locales/build/.not-empty"
+
+RUN_GRUNT=$([[ "$RUN_GRUNT" == 0 && -f "$NOT_EMPTY_JS_PATH" && -f "$NOT_EMPTY_CSS_PATH" && -f "$NOT_EMPTY_LOCALES_PATH" ]] && echo 0 || echo 1)
```
Comment (jnm) [inline setup/docker/01_setup_enketo.bash]: This should work just fine, but I'd gently suggest some line breaks to make it easier on the eyes :) Maybe:
```bash
RUN_GRUNT=$([[
    "$RUN_GRUNT" == 0 &&
    -f "$NOT_EMPTY_JS_PATH" &&
    -f "$NOT_EMPTY_CSS_PATH" &&
    -f "$NOT_EMPTY_LOCALES_PATH"
]] && echo 0 || echo 1)
```

File: setup/docker/01_setup_enketo.bash (hunk 1 above)
Comment (noliveleger) [reply setup/docker/01_setup_enketo.bash]: Totally right ;-)

File: setup/docker/01_setup_enketo.bash (hunk 2 above)
Comment (noliveleger) [reply setup/docker/01_setup_enketo.bash]: I usually break lines at 120 characters but I have to admit I abused on this one ;-)

**Pull request merged**